src/IrcConnection/IrcConnection.cs

```diff
@@ -1050,17 +1050,65 @@ private void _OnConnectionError(object sender, EventArgs e)
             } catch (ConnectionException) {
             }
         }
-        
+
+        /// <summary>
+        /// 
+        /// </summary>
+        private abstract class FiniteThread
```

Comment:
I also got over this point while working on my changes. Could you name a situation to test this case?